### PR TITLE
Fix burner wallet in forked network on UI

### LINKED ISSUE #313: Bug: Burner wallet not working with a fork of mainnet on devnet
## Issue Overview

-Run devnet with a fork of mainnet: `yarn chain --fork-network https://starknet-mainnet.public.blastapi.io/rpc/v0_7`
⁃Deployment works with burners, can't be problem of devnet
⁃Account class hash , signature mismatch. Find workaround to import same account type as the forked network.

## Proposed Solutions or Ideas
 ⁃ Try native starknet js tx on deploy script with burner account on fork
 ⁃ If previous work, check transactor file and BurnerWallet files on next js app.
 ⁃ Replicate the starknet js tx using starknet react.

## Temporary solution using current version
 Using your extension wallets on devnet:
 ⁃ Run chain with fork
 ⁃ Connect to developer network on bravos extension
 ⁃ Get your account address and use faucet to get eth
 ⁃ Deploy your account through the settings of braavos wallet, as you would do with any other account on your extension
 ⁃ You are able to interact with it as if it was a devnet burner account
Outline your proposed solution or share any ideas you have for discussing or resolving the issue.

### PR DESCRIPTION
**Pull request by jrcarlos2000**
# Burner wallet not able to submit tx on forked network

Fixes #313 

## Types of change

- [ ] Feature
- [] Bug
- [x] Enhancement

## Comments (optional)

- problem is on a fork , the account in nextjs is initialized with provider with chainId from original devnet fetched from `scaffold.config.ts`, however the actual chainId is from mainnet so transactions fail because a forked devnet chainId = mainnet chainId .
- there is an issue with wrong network, if you select devnet on `scaffold.config.ts` and you are on a fork, current fix will show `WRONG_NETWORK` on UI because it expects devnet but the chainId is from mainnet
- possible solution : `fork` flag on `scaffold.config.ts` to avoid `WRONG_NETWORK`

### CODE REVIEW

File: packages/nextjs/services/web3/stark-burner/BurnerConnector.ts [hunk 1]
```diff
@@ -59,11 +59,12 @@ export class BurnerConnector extends InjectedConnector {
       new Account(
         new RpcProvider({
           nodeUrl: this.chain.rpcUrls.public.http[0],
-          chainId: starknetChainId(this.chain.id),
+          // chainId: starknetChainId(this.chain.id), // TODO : check how this affects wrong network. can we get this from the chain directly ?
```
Comment (metalboyrick) [inline packages/nextjs/services/web3/stark-burner/BurnerConnector.ts:62]: wdym by getting from the chain directly?

Comment (0xquantum3labs) [reply packages/nextjs/services/web3/stark-burner/BurnerConnector.ts:62]: yes with the workaround of `getChainId()`

**Comment by iossocket** [comment]
Hi @jrcarlos2000, I create a pr #338 for current branch, Im thinking to add a new chain option for `scaffoldConfig`. would you mind to check it?

**Comment by iossocket** [comment]
is there something pending for this pull request?

**Review by Nadai2010 — APPROVED** [review]
For my part it works perfectly, I have tried adding Katana in supportedChains for testing and it works well.

**Review by iossocket — APPROVED** [review]

**Comment by metalboyrick** [comment]
@jrcarlos2000 pls resolve conflicts

**Pull request merged**